backend/src/apiserver/resource/resource_manager_test.go

```diff
@@ -74,6 +75,23 @@ var testWorkflow = util.NewWorkflow(&v1alpha1.Workflow{
 	Status:     v1alpha1.WorkflowStatus{Phase: v1alpha1.NodeRunning},
 })
 
+var testWorkflowValid = util.NewWorkflow(&v1alpha1.Workflow{
```

Comment:
when all the APIs expect a valid workflow, I think we should change `testWorkflow` to be valid directly.

I understand that will probably incur a lot of test expectation changes, I'd suggest constructing expectations by:

1. deep copy the testWorkflow
2. then modify attributes the test itself cares about

In this way, we should be able to reduce some test changes.

Reply:
Agree, sounds good.

Reply:
Fixed now, I actually waited to push it until I understood your comment regarding validation for uploading/creation.